bundles/org.openhab.binding.systeminfo/src/main/java/org/openhab/binding/systeminfo/internal/discovery/SystemInfoDiscoveryService.java

```diff
@@ -57,28 +58,31 @@ public SystemInfoDiscoveryService() {
     @Override
     protected void startScan() {
         logger.debug("Starting system information discovery !");
-        String hostname;
 
+        String hostname;
         try {
             hostname = getHostName();
             if (hostname.isEmpty()) {
                 throw new UnknownHostException();
             }
-            if (!hostname.matches("[" + THING_UID_VALID_CHARS + "]*")) {
-                hostname = hostname.replaceAll("[^" + THING_UID_VALID_CHARS + "]", HOST_NAME_SEPERATOR);
-            }
         } catch (UnknownHostException ex) {
             hostname = DEFAULT_THING_ID;
             logger.info("Hostname can not be resolved. Computer name will be set to the default one: {}",
                     DEFAULT_THING_ID);
         }
 
-        ThingUID computer = new ThingUID(THING_TYPE_COMPUTER, hostname);
-        thingDiscovered(DiscoveryResultBuilder.create(computer).withLabel(DEFAULT_THING_LABEL).build());
+        String thingId = hostname;
+        if (!thingId.matches("[" + THING_UID_VALID_CHARS + "]*")) {
+            thingId = thingId.replaceAll("[^" + THING_UID_VALID_CHARS + "]", HOST_NAME_SEPARATOR);
+        }
+
+        ThingUID computer = new ThingUID(THING_TYPE_COMPUTER, thingId);
+        DiscoveryResultBuilder builder = DiscoveryResultBuilder.create(computer).withLabel(DEFAULT_THING_LABEL);
+        builder.withProperty(PROPERTY_HOSTNAME, hostname).withRepresentationProperty(PROPERTY_HOSTNAME);
```

Comment:
I may miss something, but I don't see how adding this representation property makes any difference at all.
What is discovered is of thing type `THING_TYPE_COMPUTER`. What gets created in the handler factory is of thing type `THING_TYPE_COMPUTER_IMPL`. As the thing types are different, the representation property will not stop it from being discovered again.
As there is no regular scan (just a scan on demand), it doesn't make much difference.

Reply:
Right: the benefit is small, since on demand discovery only. May be i added this, due to manual triggering during tests: the thing was provided via textual configuration and triggering the discovery populated inbox again. I can remove it, if you prefer.

Reply:
I actually think it will be discovered again in any case as the actual thing type is different from the discovered thing type. So the representation property does nothing at all.

Reply:
The change does not harm, but it also does not bring anything. I don't like changing for the sake of it. You I would remove it.

Reply:
Ok. After upgrade the runtime, doesn't work anymore. Removed.

Reply:
It wouldn't work any time dynamic channel groups are added.